lib/tapioca/compilers/dsl/rails_generators.rb

```diff
@@ -0,0 +1,131 @@
+# typed: strict
+# frozen_string_literal: true
+
+begin
+  require "rails/generators"
+  require "rails/generators/app_base"
+rescue LoadError
+  return
+end
+
+module Tapioca
+  module Compilers
+    module Dsl
+      # `Tapioca::Compilers::Dsl::RailsGenerators` generates RBI files for Rails generators
+      #
+      # For example, with the following generator:
+      #
+      # ~~~rb
+      # # lib/generators/sample_generator.rb
+      # class ServiceGenerator < Rails::Generators::NamedBase
+      #   argument :result_type, type: :string
+      #
+      #   class_option :skip_comments, type: :boolean, default: false
+      # end
+      # ~~~
+      #
+      # this compiler will produce the RBI file `service_generator.rbi` with the following content:
+      #
+      # ~~~rbi
+      # # service_generator.rbi
+      # # typed: strong
+      #
+      # class ServiceGenerator
+      #   sig { returns(::String)}
+      #   def result_type; end
+      #
+      #   sig { returns(T::Boolean)}
+      #   def skip_comments; end
+      # end
+      # ~~~
+      class RailsGenerators < Base
+        extend T::Sig
+
+        BUILT_IN_MATCHER = T.let(
+          /::(ActionMailbox|ActionText|ActiveRecord|Rails)::Generators/,
+          Regexp
+        )
+
+        sig do
+          override
+            .params(
+              root: RBI::Tree,
+              constant: T.class_of(::Rails::Generators::Base)
+            )
+            .void
+        end
+        def decorate(root, constant)
+          base_class = base_class_for(constant)
+          arguments = constant.arguments - base_class.arguments
+          class_options = constant.class_options.reject do |name, option|
+            base_class.class_options[name] == option
+          end
```

Comment:
The check below could be part of `gather_constants`, but we would have to repeat the filtering in `#decorate`. This filters out class options that the parent defines, because we don't need to define them as part of the child class's RBI (since they are already part of the parent's).

I opted to only perform the filter once and include the early return below so we don't have to perform repeat work.

What do you think?